crates/db_views/src/comment_view.rs

```diff
@@ -146,7 +147,10 @@ impl CommentView {
   }
 }
 
-pub struct CommentQueryBuilder<'a> {
+#[derive(TypedBuilder)]
+#[builder(field_defaults(default))]
+pub struct CommentQuery<'a> {
+  #[builder(!default)]
```

Comment:
The docs seem to be saying the opposite of what you're doing here. https://idanarye.github.io/rust-typed-builder/typed_builder/derive.TypedBuilder.html

The Options should all use builder default ( so that they default to None), while the mandatory fields should have nothing.

Reply:
Yes that is what its doing, just the other way round. Instead of specifying `#[builder(!default)]` for every optional field, i put `#[builder(field_defaults(default))]` which means that all fields start with the default value (None). Then mandatory fields have `#[builder(!default)]` (not default), so that they need to be initialized manually. Its documented further down on that page.